tools/benchmark/macros/Cargo.toml

```diff
@@ -11,5 +11,10 @@ version = "0.1.0"
 [dependencies]
 dhat = { version = "0.2", optional = true }
 
+# This cfg originates in dlmalloc/lib.rs
+[target.'cfg(any(target_os = "linux", target_os = "macos", target_arch = "wasm32"))'.dependencies]
+dlmalloc = { version = "0.2", optional = true, features = ["global"] }
+
 [features]
 benchmark_memory = ["dhat"]
+rust_global_allocator = ["dlmalloc"]
```

Comment:
Question: Why is this called `rust_global_allocator`? Could it be possible that we want other global allocator configurations? Would it be better to call this something like `valgrind_allocator`?

Reply:
What @dminor said.  The point is to remove the system dependency.  There's no particular reason that this is `dlmalloc` (it could be any other Rust-based allocator), and it could be useful outside of Valgrind.

Reply:
Ah, I _think_ the dhat benchmark sets its own custom global allocator too.

Reply:
Indeed it does, which is why these two features are mutually exclusive.

I tried just using the dhat global allocator, but it had too much overhead for it to be useful in performance benchmarks.